packages/headless/src/features/configuration/configuration-slice.ts

```diff
@@ -128,4 +129,8 @@ export const configurationReducer = createReducer(
         state.analytics.originLevel2 =
           action.payload.tab || state.analytics.originLevel2;
       })
+      .addCase(analyticsDescription, (state, action) => {
+        state.analytics.actionCause = action.payload.actionCause;
+        state.analytics.customData = action.payload.customData;
```

Comment:
Could these parameters be a part of the analytics action payload? So, calling `logSearchboxSubmit` would return an object wuth shape:

```
{
  type: 'analytics/search-box/submit',
  payload: {
    actionCause: 'search-box-submit',
    customData: {...},
    ...
  },
}
```

Writing/reading from state will work most of the time, but the indirection makes it possible for the values to be unintentionally overwritten by another dispatch.